### PR TITLE
Delayed resolve or reject of Promise until after callback completes

### LINKED ISSUE #9: Listener resolves promise too early
We should change `createListenForActions` to delay resolve or reject until after the callback completes, otherwise we risk missing errors or dispatched actions that occur after the last expected action is caught but before the callback returns

### PR DESCRIPTION
**Pull request by noisecapella**
Fixes #9

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -83,18 +83,31 @@ function createListenForActions(store, stateFunc, actionListFunc, blacklistedAct
...
+            // wait until after callback() completes to resolve so we can detect thrown errors
+            // and actions which shouldn't have been dispatched
+            resolver();
+          }
         });
 
         callback();
+        callbackRan = true;
+        // If already resolved, this should get ignored
+        resolver();
```
Comment (alicewriteswrongs) [inline src/index.js:110]: I can't pretend to fully understand everything going on here, but this logic looks straightforward enough - we subscribe to the store with a function that will run the resolver only after we set `callbackRan` to `true`, then we call the callback and set that boolean.

Before we were just setting that same subscription and then running the callback, which results in a race condition where if the expected actions are dispatched before we hit an error in the callback we resolve the promise anyway and the error is swallowed.

is that accurate?

**Comment by alicewriteswrongs** [comment]
one question to make sure I'm understanding what we're doing, generally LGTM tho! (tentative :+1:)

File: src/index.js (hunk 1 above)
Comment (noisecapella) [inline src/index.js:110]: Yes, pretty much

Comment (alicewriteswrongs) [inline src/index.js:110]: cool, just wanted to check my understanding,

**Comment by alicewriteswrongs** [comment]
more confident  :+1: !

**Pull request merged**